anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -559,6 +563,424 @@ def group_by_name(self, group_name):
+        )
+
+
+class GrypeDBFeed(AnchoreServiceFeed):
+    """
+    AnchoreServiceFeed used to sync Grype DB data.
+
+    :param metadata: FeedMetadata instance to use to store stateful information about this feed
+    :type metadata: Optional[FeedMetadata], defaults to None
+    """
+
+    __feed_name__ = "grypedb"
+    _cve_key = None
+    __group_data_mappers__ = {}
+
+    def __init__(self, metadata: Optional[FeedMetadata] = None):
+        """
+        Constructor method.
+        """
+        # The catalog_client is stored with the instance here to to avoid breaking Liskov Substitution Principle.
+        # Signatures for overridden methods from superclass should not change in subclass.
+        self._catalog_svc_client: Optional[CatalogClient] = None
+        super().__init__(metadata=metadata)
+
+    @property
+    def _catalog_client(self) -> CatalogClient:
+        """
+        Getter for catalog client
+
+        :return: catalog client instance
+        :rtype: CatalogClient
+        """
+        if not isinstance(self._catalog_svc_client, CatalogClient):
+            logger.debug(
+                "Catalog Client not initialized in GrypeDBFeed. Initializing..."
+            )
+            self._catalog_svc_client = internal_client_for(CatalogClient, userId=None)
+        return self._catalog_svc_client
+
+    def _find_match(
+        self, db: Session, checksum: Optional[str] = None, active: Optional[bool] = None
+    ) -> Query:
+        """
+        Utility Method, queries GrypeDBMetadata and optionally filters on checksum or active attribute.
+
+        :param db: sqlalchemy database session
+        :type db: Session
+        :param checksum: checksum of the grype db file
+        :type checksum: Optional[str], defaults to None
+        :param active: whether or not to filter on the active record
+        :type active: Optional[bool], defaults to None
+        :return: sqlalchemy query object
+        :rtype: Query
+        """
+        results = db.query(GrypeDBMetadata)
+        if checksum:
+            results = results.filter(GrypeDBMetadata.checksum == checksum)
+        if not isinstance(active, type(None)):
+            results = results.filter(GrypeDBMetadata.active == active)
+        return results
+
+    def record_count(self, group_name: str, db: Session) -> int:
+        """
+        Returns number of records present in the database for a given group (GrypeDBMetadata records)
+
+        :param group_name: name of the group
+        :type group_name: str
+        :param db: sqlaclhemy database session
+        :type db: Session
+        :return: number of records
+        :rtype: int
+        """
+        try:
+            return self._find_match(db).count()
+        except Exception:
+            logger.exception(
+                "Error getting feed data group record count in package feed for group: {}".format(
+                    group_name
+                )
+            )
+            raise
```

Comment:
This is not doing much other than adding the exception logging with a custom message. I would suggest avoiding the broad try/except and prefer adding a logging.debug or logging.info that explains what the action (getting feed data) is about

Reply:
Good point. I actually just copied that try except from the sibling subclass, VulnerabilityFeed, and never gave it a second thought. I'll just throw out the try/except.